### ISSUE TITLE
Code fix: SA1405 (DebugAssertMustProvideMessageText)

### ISSUE THREAD

**Issue by sharwell** [DotNetAnalyzers/StyleCopAnalyzers#657]
Implement a code fix for #94.

A candidate for the generated string is rendering the assertion expression to a string literal.

**Comment by vweijsters** [comment]
Do we really want a code fix here? The goal is that the developer provides a meaningful message, something which will be impossible to generate. I would personally prefer to have no code fix over having a code fix that generates meaningless boilerplate text (something that will not improve code quality)

(Same holds for #658)

**Comment by sharwell** [comment]
My initial thought was a reasonable message for the following:

```csharp
Debug.Assert(x != null || Foo());
```

Would be: `Assertion failed: x != null || Foo()`

**Comment by vweijsters** [comment]
I think that message has no value, other than eliminating the useful warning. I would prefer forcing developers to enter a message that is more meaningful than that. Of course it is impossible to really enforce that from a tool, but I would opt for not providing an easy way out.

**Comment by sharwell** [comment]
Remember that `Debug.Assert` is typically only even evaluated in a debug build. Personally if I can see the assertion that failed, I already have a very good idea what the problem was.

**Comment by vweijsters** [comment]
I agree, I normally disable SA1405 because it has limited value.

**Comment by dlemstra** [comment]
I would also think there should be no CodeFix for this one. The developer should add their own message if this rule is enabled. Maybe this issue should be closed?
